.circleci/config.yml

```diff
@@ -233,24 +237,75 @@ jobs:
           stage: "production"
 
 workflows:
+  check-on-feature-branch:
+    jobs:
+      - check-code-formatting:
+          context: api-nuget-token-context
+          filters:
+            branches:
+              ignore: release, master
+      - sonarcloud-scan:
+          context: SonarCloud
+          filters:
+            branches:
+              ignore: release, master
+      - build-and-test:
+          context:
+            - api-nuget-token-context
+          filters:
+            branches:
+              ignore: release, master
+      - assume-role-development:
+          context: api-assume-role-housing-development-context
+          filters:
+            branches:
+              ignore: release, master
+      - terraform-init-and-plan-development:
+          requires:
+            - assume-role-development
+      - terraform-compliance-development:
+          requires:
+            - terraform-init-and-plan-development
+
+
   check-and-deploy-development:
```

Comment:
I made this only run on development. It's clearer and also lets the check-on-feature-branch be separate